TLM/TLM/Manager/Impl/JunctionRestrictionsManager.cs

```diff
@@ -338,8 +338,9 @@ public class JunctionRestrictionsManager
                     segmentId,
                     startNode,
                     ref node)) {
-                bool res = (node.m_flags & (NetNode.Flags.End | NetNode.Flags.OneWayOut)) !=
-                           NetNode.Flags.None;
+                // vanilla one-way or no other segment with matching vehicle categories
+                bool res = (node.m_flags & (NetNode.Flags.End | NetNode.Flags.OneWayOut)) != NetNode.Flags.None ||
+                           ((node.m_flags2 & NetNode.Flags2.PedestrianStreetTransition) != 0 &&!HasOtherSegmentsWithCategories(segmentId, ref node));
```

Comment:
`NeNode.Flags2.RegularRoadEnd` is not useful here is it?

Reply:
Nope. we need opposite. I tried that before. 
The code works just like described in the comment. If there is a sudden change of vehicle categories it will automatically allow U-turn. In vanilla U-turns are handled only in pathfinding and we have custom code and also UI so needs to be like that. E.g. when there is a new road with pedestrian and bus only and it is connected to pedestrian only. It will have bollards on pavement side so it's not `RegularRoadEnd` unfortunately (new pedestrian+bus is not even regular road 😂 )